### ISSUE TITLE
Create /protoboards endpoint that serves protoboards from file system.

### ISSUE THREAD

**Issue by ebb-tide** [influxdata/platform#1877]
only need `/GET/protoboards` and `GET /protoboards/ID`
exists in 1.x

**Comment by goller** [comment]
I think we'll do something like `/examples` as we want to support alert prototypes.

Additionally, @desa had an idea around using links or aliases to other locations.  This would show that the dashboard is "aliased" to some other place (perhaps an item in /example or even something in github).

If one wanted, you could convert that aliased/linked dashboard into your own local copy.

**Comment by ebb-tide** [comment]
what is this blocked on? It would be really nice to have it in alpha

**Comment by goller** [comment]
@ebb-tide We had talked about getting people together for a kick-off before starting this.  Blocked is such a strong word, but, "waiting" until we kick-off.

**Comment by desa** [comment]
For the alpha, we've decided to add the following endpoints


**GET /api/v2/prototypes**
```json
{
  "prototypes": [
    {
      "links": {
        "dashboards": "...",
        "tasks": "...",
        "self": "..."
      },
      "name": "...",
      "id": "..."
    },
    {
      "links": {
        "dashboards": "...",
        "tasks": "...",
        "self": "..."
      },
      "name": "...",
      "id": "..."
    }
  ]
}
```

**GET /api/v2/prototype/:id/dashboards**
```json
{
    "meta": {
        "chronografVersion": "2.0"
    },
    "dashboard": {
        "name": "Name this dashboard",
        "description": "",
        "cells": ["..."], 
    }
}
```

The specific resource name is still up for discussion. Proposed names are `prototypes`, `protos`, `templates`, and `examples`.

It is intended that the prototype will allow for multiple types of prototype structures (things like `tasks`, `dashboards`, `alerts`, `scapers`, etc).

The intention of this structure is to prototypes/examples/templates to be structured into logical groups of related concepts. As an example, they may be structured around a telegraf plugin. 

These resources will be statically defined and built into the binary for the alpha.

Future implementations will 

* add support for other types of templates/prototypes/examples (e.g. tasks, alerts, and scrapers).
* add support for remotely defined/registered templates/prototypes (e.g. `dashboard.land`).

**Comment by desa** [comment]
@ebb-tide @goller @russorat Does the above seem like a good plan?

**Comment by russorat** [comment]
@desa i would only ask that we use something other than chronografVersion in the meta field. influxdbVersion maybe.

Also, in the future, the UI will instantiate multiple of these. So it might be helpful (not alpha) to have the endpoint accept a list of ids and instantiate multiple.

Also, will a dashboard (or other thing) know which prototype it was created from? Do we need to store that somewhere?

**Comment by desa** [comment]
>  i would only ask that we use something other than chronografVersion in the meta field. influxdbVersion maybe.

We can add anything you'd like to meta

> Also, in the future, the UI will instantiate multiple of these. So it might be helpful (not alpha) to have the endpoint accept a list of ids and instantiate multiple.

👍 

> Also, will a dashboard (or other thing) know which prototype it was created from? Do we need to store that somewhere?

Yes we'll store that on the dashboard in the metadata.
